documentation/src/main/asciidoc/ch11.asciidoc

```diff
@@ -694,17 +694,25 @@ mechanism to register additional constraint definitions. All you have to do is t
 _javax.validation.ConstraintValidator_ to _META-INF/services_. In this service file you list the
 fully qualified classnames of your constraint validator classes (one per line). Hibernate Validator
 will automatically infer the constraint types they apply to.
-See <<example-using-service-file-for-constraint-definitions>> for an example.
+See <<example-using-service-file-for-constraint-definitions,Constraint definition via service file>>
+for an example.
 
 [[example-using-service-file-for-constraint-definitions]]
-.Using _META-INF/services/javax.validation.ConstraintValidator_ for constraint definitions
+._META-INF/services/javax.validation.ConstraintValidator_
+
 ====
 [source]
 ----
+# Assuming a custom constraint annotation @org.mycompany.CheckCase
 org.mycompany.CheckCaseValidator
 ----
 ====
 
+To contribute default messages for your custom constraints, place a _ValidationMessages.properties_
```

Comment:
> How about having one named bundle per archive instead which then is "merged" (e.g. similar to service loader files)?

I thought about this, but in the end settled for this. Let me give you some pro and cons for this approach:
Pro:
- Simple, no need for yet another contract/api
- We use `ValidationMessages.properties` consistently. This makes the connection to Bean Validation obvious. It also prevent proliferation of names when everyone tries to name the properties file something else
- It automatically creates a natural namespace. Using a new contract, people could for example put a `ValidationMessages.properties` into the root of their jar. This would then conflict with the users `ValidationMessages.properties`

Con:
- If you have custom constraint spread over multiple packages, you need to provide multiple message bundles. However, I expect this to be the exception. 

> I think that'd be a bit easier to comprehend/maintain

Don't you think this rule is pretty straight forward?

>  It'd need to have a different name than ValidationMesssages though

Kind of what I would like to prevent.